CONTRIBUTING.md

```diff
@@ -30,3 +30,66 @@ Use culturally neutral language:
 
   What you may consider well-known may be entirely obscure and distracting to people from different backgrounds.
 
+# Target audience
+
+The main target audience are software developers with basic experience on the Linux command line.
+Specific prerequisite knowledge and skills should always be stated, see [Structure](#structure) for details.
+
+These guides should work for autodidacts who do not have access to hand-holding or personal training, nor possibility for learning through osmosis.
+
+# Structure
+
+Each guide should have the following structure:
+
+- What will you learn? (problem statement and learning goals)
+- What do you need? (prerequisites)
+
+  The material should always be written such that the stated prerequisites are sufficient to achieve learning goals.
+
+  - previous chapters
+  - domain-specific skills or knowledge
+
+- How long does it take? (reading time)
+
+  This will depend on the learner's pre-existing knowledge and proficiency.
+  Note here how optional skills or knowledge may influence reading time.
+
+- (optional) How easy is it? (level of difficulcy)
+
+  - Some things are trivial and fun.
+  - Some things require disciplined learning to get productive.
+  - Some things do not work properly, have confusing interfaces, or inadequate documentation.
+
+- How to do it? (instructions)
+
+  Depending on how well a use case is supported, we can provide
+  - instructions and examples
+  - links to known-good external resources, with summaries
+  - overview of available tools, and their state of maturity and maintenance
+  - overview of ideas, and state of community discussion.
+
+- What did you learn? (assessment)
+
+  To further improve the guides, the tools and their documentation, we assess your learning success.
+  You can also ask questions, and leave feedback or suggestions.
+
+- (optional) How does it work? (explanation)
+
+  For the curious: a brief explanation of the mechanism behind the solution Nix offers.
+
+- How to continue from here? (next steps)
+
+  Depending on how well a use case is explored, you can proceed to
+  - other chapters
+  - reference manuals
+  - ongoing community discussion.
+  - additional resources
```

Comment:
What about also adding a few optional exercises to the reader?
To give meat to the learner that asks himself: "with this, what can I do to train and check I understood the guide correctly?"